src/test/script_p2sh_tests.cpp

```diff
@@ -212,24 +212,51 @@ BOOST_AUTO_TEST_CASE(is)
     p2sh << OP_HASH160 << ToByteVector(dummy) << OP_EQUAL;
     BOOST_CHECK(p2sh.IsPayToScriptHash());
 
-    // Not considered pay-to-script-hash if using one of the OP_PUSHDATA opcodes:
+    // Same test as above, but use string interface
     std::vector<unsigned char> direct = {OP_HASH160, 20};
-    direct.insert(direct.end(), 20, 0);
+    direct.insert(direct.end(), 20, 0); // this is 160bit dummy
     direct.push_back(OP_EQUAL);
     BOOST_CHECK(CScript(direct.begin(), direct.end()).IsPayToScriptHash());
+
+    //---------><----- cut here
+    // orig version fixed by re-adding OP_PUSHDATA2/4
```

Comment:
Can you explain what `cut here` and `orig version` means? I think those are confusion to test readers.

Reply:
Oh, sorry my commit comment was a bit short.

I've made two versions of the fix since I didn't know what you prefer.

The orig version is clearer in what it fixes.
The block below that is perhaps less code, though not perhaps the quickest to understand at-a-glance.